File: c-sharp/src/HelloWorld/HelloWorld.cs
using System;

namespace HelloWorld
{
	class HelloWorld
	{

		public static readonly string helloTemplate;
		public static readonly string defaultUser;

		static UserMessage()
		{
			helloTemplate = "Hello {0}!";
			defaultUser = "World";
		}

		public static string SayHello()
		{
			return SayHello(defaultUser);
		}

		public static string SayHello(string name)
		{
			return String.Format(helloTemplate,name);
		}

		static void Main(string[] args)
		{
			if (args.length > 0)
			{
				for (int i = 0; i < args.Length; i++)
				{
					Console.WriteLine(SayHello(s));
				}
			}
			else
			{
				Console.WriteLine(SayHello());
			}

		}
	}
}

File: java/src/main/java/z594/helloworld/HelloWorld.java
package z594.helloworld;

public final class HelloWorld {

	public static final String helloTemplate = "Hello %s!";
	public static final String defaultUser = "World";

	public static String sayHello() {
		return sayHello(defaultUser);
	}

	public static String sayHello(String name) {
		return String.format(helloTemplate,name);
	}

	public static void main(String[] args) {
		if (args.length > 0) {
			for (String s: args) {
				System.out.println(sayHello(s));
			}
		} else {
			System.out.println(sayHello());
		}
	}

}
